pandas-stubs/core/reshape/tile.pyi

```diff
@@ -1,13 +1,212 @@
+from typing import (
+    Literal,
+    Sequence,
+    overload,
+)
+
+import numpy as np
+from pandas import (
+    Categorical,
+    Float64Index,
+    Index,
+    Int64Index,
+    IntervalIndex,
+    Series,
+)
+
+from pandas._typing import (
+    Label,
+    npt,
+)
+
+@overload
 def cut(
-    x,
-    bins,
+    x: Index | npt.NDArray | Sequence[int] | Sequence[float],
+    bins: int | Series | Int64Index | Float64Index | Sequence[int] | Sequence[float],
     right: bool = ...,
-    labels=...,
-    retbins: bool = ...,
+    *,
+    labels: Literal[False],
+    retbins: Literal[True],
     precision: int = ...,
     include_lowest: bool = ...,
-    duplicates: str = ...,
-): ...
+    duplicates: Literal["raise", "drop"] = ...,
+    ordered: bool = ...,
+) -> tuple[npt.NDArray[np.intp], npt.NDArray]: ...
+@overload
+def cut(
+    x: Index | npt.NDArray | Sequence[int] | Sequence[float],
+    bins: IntervalIndex,
+    right: bool = ...,
+    *,
+    labels: Literal[False],
+    retbins: Literal[True],
+    precision: int = ...,
+    include_lowest: bool = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+    ordered: bool = ...,
+) -> tuple[npt.NDArray, IntervalIndex]: ...
```

Comment:
first part of tuple has to be integers `np.NDArray[np.intp]`

Reply:
Fixed.